### ISSUE TITLE
Keras Callbacks documentation for on_train_batch_end vs the actual code of ModelCheckpoint

### ISSUE THREAD

**Issue by durandg12** [tensorflow/tensorflow#29801]
## URL(s) with the issue:

[https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/callbacks/Callback#on_train_batch_end](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/callbacks/Callback#on_train_batch_end)

### Parameters defined

The parameter section of `on_train_batch_end` reads, in part,

> `logs`: dict. Metric results for this batch

So the `logs` argument is expected to only contain metrics when this method is called. This is what I assumed when I coded my own custom training function.

This is different than the `logs` argument which is expected by `on_train_batch_begin` for example, which is:

> `logs`: dict. Has keys `batch` and `size` representing the current batch number and the size of the batch.

Now if we look at [the code](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/callbacks.py) of `ModelCheckpoint`, we see that is has an `on_batch_end` method (which is called by `on_train_batch_end`) which calls the `size` key of `logs`. This is line 950:
`self._samples_seen_since_last_saving += logs.get('size', 1)`

So the documentation of `on_train_batch_end` is not correct as to what is the dict expected by the method. And this has an impact when we want to create custom training functions that work well with Keras Callbacks.

**Comment by jvishnuvardhan** [comment]
@durandg12 I think the document was updated since `TF2.0` and I cannot see that issue (self._samples_seen_since_last_saving += logs.get('size', 1)) anymore.  

Please check the recent code in `master` branch https://github.com/keras-team/keras/blob/master/keras/callbacks.py#L572-L880

Can you please verify once and close the issue if this was resolved for you.

**Comment by durandg12** [comment]
`ModelCheckpoint` no longer has a `on_batch_end` method and its `on_train_batch_end` method don't use any key of `logs`. So it seems like the issue is indeed closed.
